SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityS3UploaderDriver.java

```diff
@@ -515,4 +529,27 @@ private boolean isS3MetadataFile(Path filename) {
 
     return true;
   }
+
+  private boolean canCreateImmediateUploader(S3UploadMetadata metadata) {
+    try {
+      if (lock.tryLock(400, TimeUnit.MILLISECONDS)) {
+        if (this.immediateUploadMetadata.contains(metadata)) {
+          LOG.debug("Stopped from creating a duplicate immediate uploader.");
```

Comment:
Let's make these statements slightly more useful by adding the actual metadata to them. Like:
```java
LOG.debug("Already have an immediate uploader for {}", metadata);
```